src/core/ready.js

```diff
@@ -5,12 +5,13 @@ define( [
 ], function( jQuery, document ) {
 
 // The deferred used on DOM ready
-var readyList;
+var readyList = jQuery.Deferred(),
+	readyPromise = readyList.promise();
 
 jQuery.fn.ready = function( fn ) {
 
 	// Add the callback
-	jQuery.ready.promise().done( fn );
+	readyPromise.done( fn );
```

Comment:
I actually tried switching to `then`, but didn't get the behavior I expected. I'll keep digging.